src/cache/core/types/Cache.ts

```diff
@@ -104,12 +161,197 @@ export declare namespace Cache {
     ) => any;
   }
 
-  export import DiffResult = DataProxy.DiffResult;
-  export import ReadQueryOptions = DataProxy.ReadQueryOptions;
-  export import ReadFragmentOptions = DataProxy.ReadFragmentOptions;
-  export import WriteQueryOptions = DataProxy.WriteQueryOptions;
-  export import WriteFragmentOptions = DataProxy.WriteFragmentOptions;
-  export import UpdateQueryOptions = DataProxy.UpdateQueryOptions;
-  export import UpdateFragmentOptions = DataProxy.UpdateFragmentOptions;
-  export import Fragment = DataProxy.Fragment;
+  export interface ReadQueryOptions<
+    TData,
+    TVariables extends OperationVariables,
+  > {
+    /**
+     * The GraphQL query shape to be used constructed using the `gql` template
+     * string tag from `graphql-tag`. The query will be used to determine the
```

Comment:
```suggestion
     * string. The query will be used to determine the
```

Thoughts on removing this? We suggest importing from `@apollo/client` in most cases, so not sure if this is confusing or not.

Reply:
Honestly, even the rest of the comment goes into irrelevant details. But that's probably a good one for another sweeping pass :)